be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -253,30 +253,37 @@ Status SegmentIterator::_get_row_ranges_from_conditions(RowRanges* condition_row
     _opts.stats->rows_bf_filtered += (pre_size - condition_row_ranges->count());
 
     RowRanges zone_map_row_ranges = RowRanges::create_single(num_rows());
-    std::map<int, std::vector<CondColumn*>> column_delete_conditions;
-    // zone map will use delete conditions
-    for (auto& delete_condition : _opts.delete_conditions) {
-        for (auto& column_condition : delete_condition->columns()) {
-            cids.insert(column_condition.first);
-            std::vector<CondColumn*>& conditions = column_delete_conditions[column_condition.first];
-            conditions.emplace_back(column_condition.second);
-        }
-    }
+    // second filter data by zone map
     for (auto& cid : cids) {
         // get row ranges by zone map of this column,
         RowRanges column_row_ranges = RowRanges::create_single(num_rows());
-        CondColumn* column_cond = nullptr;
-        if (_opts.conditions != nullptr) {
-            column_cond = _opts.conditions->get_column(cid);
-        }
+        CondColumn* column_cond = _opts.conditions->get_column(cid);
```

Comment:
`_opts.conditions` may be `nullptr` here.
You need to check it before use, like the old way.

Reply:
done